### PR TITLE
Test/rearranging and fixing tests

### LINKED ISSUE #50: ToThrow messages not decleared
I have just seen an failure where we stopped performing a test correctly because a folder wasn't renamed correctly.

However, we did not catch it as the test case was expected `toThrow()`. However, the expected case was `toThrow()` with a specific error and after the incorrect rename the error message was different to expected and the test was no longer looking after the correct code branch.

I am proposing that for all tests which have a `toThrow` we should require the specific error message we are expecting to improve relatiblity. 

We should see if we can enforce this with a eslint rule.

### PR DESCRIPTION
**Pull request by PKWadsy**
This PR makes several improvements on top of PR #47 regarding tests

- Reorganises tests
- Adds absolute imports
- Updates ZodValidator to work with any time which conforms to a certain shape. This can be made even more generic but I think we can leave it for now.
- Tests for specific error types to be thrown. Closes #50 
- Adds a new testing script `test:unit` to run tests with coverage on non-integration tests.
- Coverage now includes all src files, not just files which were tested

### CODE REVIEW

**Review by joshuabrownenz — CHANGES_REQUESTED** [review]
Minors. 

I really like the splitting up of unit and integration. I don't know if I'm over complicating it but it would be quite cool to have seperate coverage metrics for unit and integration?

File: src/zod/models/ZodCompatibleRequest.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
```
[the remark this answers is not in the document]
Comment (PKWadsy) [reply src/zod/models/ZodCompatibleRequest.ts:2]: Could be unknown. But I don't think it matters too much as it is never used as a concrete type anyways

File: src/core/endpoint-tree/EndpointTree.ts [hunk 2]
```diff
@@ -238,3 +239,14 @@ export class EndpointTreeNode<
     }
   }
 }
+
```
[the remark this answers is not in the document]
Comment (PKWadsy) [reply src/core/endpoint-tree/EndpointTree.ts:242]: Made an issue

File: src/core/endpoint-builder/handlerWithValidators.ts [hunk 3]
```diff
@@ -55,3 +56,10 @@ export function wrapHandlerWithValidationLayer<
 
   return handler;
 }
+
```
[the remark this answers is not in the document]
Comment (PKWadsy) [reply src/core/endpoint-builder/handlerWithValidators.ts:59]: Made an issue

File: src/zod/models/ZodCompatibleRequest.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshuabrownenz) [reply src/zod/models/ZodCompatibleRequest.ts:2]: I meant. Should it be Zoe schema any

[the remark this answers is not in the document]
Comment (joshuabrownenz) [reply src/zod/models/ZodCompatibleRequest.ts:2]: Oh. This is the validator input with the schemas is it

**Comment by joshuabrownenz** [comment]
Happy for you to merge when you are ready. I won't be available for the rest of the day to give another review

**Pull request merged**